x/nexus/keeper/rate_limit.go

```diff
@@ -0,0 +1,130 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	"github.com/axelarnetwork/axelar-core/utils/events"
+	"github.com/axelarnetwork/axelar-core/utils/key"
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+	"github.com/axelarnetwork/utils/funcs"
+)
+
+// RateLimitTransfer applies a rate limit to transfers, and returns an error if the rate limit is exceeded
+func (k Keeper) RateLimitTransfer(ctx sdk.Context, chain exported.ChainName, asset sdk.Coin, outgoing bool) error {
+	rateLimit, found := k.getRateLimit(ctx, chain, asset.Denom)
+	if !found {
+		return nil
```

Comment:
why return nil and not an error?

Reply:
If the rate limit is not set, it is unbounded.

Reply:
Add a comment plz

Reply:
do we need to handle it this way? Could we set it explicitly to unbounded instead?

Reply:
Existing chain/assets combinations will need a migration in that case, and the proto will be a bit more complicated. This seems like a simple alternative, in line with how we do it elsewhere.